### PR TITLE
Improve trailing slash handling

### PR DESCRIPTION
**Pull request by TeslaCN**
Ref #1284 

Trailing slash is a slash at the end of uri, like "/app/job/".
If trailing slash sensitive, "/app/job" and "/app/job/" are not equivalent. Otherwise, "/app/job" and "/app/job/" will be mapped to a same handler.
Trailing slash in insensitive by default. Developers can enable trailing slash sensitive by configuring NettyResfulServiceConfiguration.

Changes proposed in this pull request:
- Add option "trailingSlashSensitive".
- Complete testcases.

### CODE REVIEW

File: elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/serializer/CustomTextPlainResponseBodySerializer.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.elasticjob.restful.serializer;
+
+import io.netty.handler.codec.http.HttpHeaderValues;
+
+import java.nio.charset.StandardCharsets;
+
+public final class CustomTextPlainResponseBodySerializer implements ResponseBodySerializer {
```
Comment (Technoboy-) [inline elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/serializer/CustomTextPlainResponseBodySerializer.java:27]: Miss java doc

File: elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpClient.java [hunk 2]
```diff
@@ -66,8 +65,11 @@ protected void initChannel(final Channel ch) throws Exception {
                                 .addLast(new SimpleChannelInboundHandler<FullHttpResponse>() {
                                     @Override
                                     protected void channelRead0(final ChannelHandlerContext ctx, final FullHttpResponse httpResponse) throws Exception {
-                                        consumer.accept(httpResponse);
-                                        countDownLatch.countDown();
+                                        try {
+                                            consumer.accept(httpResponse);
+                                        } finally {
+                                            countDownLatch.countDown();
```
Comment (Technoboy-) [inline elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpClient.java:73]: Why need CountDownLatch? Maybe I have to clone the branch for deeply reviewing .

File: elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/controller/IndexController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+package org.apache.shardingsphere.elasticjob.restful.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shardingsphere.elasticjob.restful.Http;
+import org.apache.shardingsphere.elasticjob.restful.RestfulController;
+import org.apache.shardingsphere.elasticjob.restful.annotation.Mapping;
+
+@Slf4j
+public class IndexController implements RestfulController {
```
Comment (Technoboy-) [inline elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/controller/IndexController.java:26]: Add final ?

File: elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/NettyRestfulServiceTest.java [hunk 4]
```diff
@@ -82,43 +84,59 @@ public void assertRequestWithParameters() {
             assertThat(jobPojo.getGroup(), is("myGroup"));
             assertThat(jobPojo.getName(), is("myJob"));
             assertThat(jobPojo.getDescription(), is(description));
-        }, 10000L);
+        }, TESTCASE_TIMEOUT);
     }
     
-    @Test(timeout = 10000L)
-    public void assertCustomExceptionHandler() {
+    @Test(timeout = TESTCASE_TIMEOUT)
```
Comment (Technoboy-) [inline elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/NettyRestfulServiceTest.java:91]: Recommend to use @SneakyThrows for clean code.

File: elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpClient.java (hunk 2 above)
Comment (TeslaCN) [reply elasticjob-infra/elasticjob-restful/src/test/java/org/apache/shardingsphere/elasticjob/restful/pipeline/HttpClient.java:73]: > Why need CountDownLatch? Maybe I have to clone the branch for deeply reviewing .

The HTTP client for unit test is implemented via Netty, which handle http response asynchronously. So I use CountDownLatch to wait for http response consumer finish.

**Comment by TeslaCN** [comment]
CI [https://travis-ci.org/github/apache/shardingsphere-elasticjob/builds/720907655](https://travis-ci.org/github/apache/shardingsphere-elasticjob/builds/720907655) failed because of #1405

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**